pyogrio/geopandas.py

```diff
@@ -254,6 +255,11 @@ def write_dataframe(
     layer_options : dict, optional
         Layer creation option (format specific) passed to OGR. Specify as
         a key-value dictionary.
+    append : bool, optional (default: False)
```

Comment:
Something else: I like `append=True/False`, but note that geopandas / fiona uses `mode="w"/"a"` (but fiona has this because it mimics the `open(..)` API, and in geopandas we can easily map the keyword for compat; we currently even explicitly raise an error for `mode="a"` on the geopandas side in case of pyogrio engine)

Reply:
I see that pandas supports the same modes as the `open()` API for things like `to_csv()`.  So we may want to be consistent but only support a subset: `write_dataframe` would never support `mode="r"`, for instance.

I'm not sure what the meaning of `mode="r+"` would be here, since we don't expose a stateful API, and we don't (yet) let you update existing records an existing data layer.

In which case, `read_dataframe` would only ever use `mode="r"` (so parameter is not needed), and `write_dataframe` could only ever use `mode="w"` or `mode="a"` - which ends up being an awful lot like our boolean `append`.

Would be nice not to change this API later, so I'm certainly open to using `mode` in order to make this more compatible with GeoPandas if that is what you'd like here.

Reply:
Personally, for a direct user of pyogrio, I find `append=True` the nicer and easier API.